pkg/sql/compile/compile.go

```diff
@@ -1307,6 +1307,24 @@ func (c *Compile) compileTableScanWithNode(n *plan.Node, node engine.Node) *Scop
 		},
 	}
 	s.Proc = process.NewWithAnalyze(c.proc, c.ctx, 0, c.anal.Nodes())
+
+	// Register runtime filters
+	// XXX currently we only implement runtime filter on single CN
+	if len(c.cnList) == 1 {
```

Comment:
i think use https://github.com/matrixorigin/matrixone/blob/main/pkg/vm/pipeline/pipeline.go#L59 may be a better choice